File: Py-Solutions/Problem-03.py
# What is the largest prime factor of the number 600851475143?
import math

def isPrime(num):
  return sum([i for i in range(1, int(math.sqrt(num)) + 1) if num % i == 0]) == 1

def largestPrimeFactor(num):
  return [i for i in range(2, int(math.sqrt(num)) + 1) if num % i == 0 and isPrime(i)][-1]

print(largestPrimeFactor(600851475143))

File: JS-Solutions/Problem-03.js
// What is the largest prime factor of the number 600851475143?

const isPrime = (num) => {
  if (num < 2) {
    return false;
  }
  if (num == 2) {
    return true;
  }
  for(let i=2;i<=Math.sqrt(num)+1;i++){
    if(num % i == 0){
      return(false);
    }
  }
  return(true);
}

const largestPrimeFactor = (num) => {
  let largest = 0;
  for(let i=2;i<=Math.sqrt(num)+2;i++){
    if(num % i == 0 & isPrime(i)){
      largest = i;
    }
  }
  return(largest);
}


console.log(largestPrimeFactor(600851475143));
